mixnet/node/src/lib.rs

```diff
@@ -92,16 +84,13 @@ impl MixnetNode {
     }
 
     async fn handle_final_payload(
-        mut socket: TcpStream,
         _private_key: PrivateKey,
         client_address: SocketAddr,
+        body: Body<'_>,
     ) -> Result<(), Box<dyn Error>> {
         // TODO: Reuse a conn instead of establishing ad-hoc conns
         let mut client_stream = TcpStream::connect(client_address).await?;
-
-        // TODO: Decrypt the final payload using the private key
-        let _ = tokio::io::copy(&mut socket, &mut client_stream).await?;
-
+        body.write(&mut client_stream).await?;
```

Comment:
In the previous implementation, this function doesn't do `write_u8` because this just tosses the final payload to the mixnet client, and the mixnet client was implemented to receive only final payloads (not sphinx packets). 
But, I'm thinking of changing the mixnet client to use a common protocol. Let me think.

Reply:
I just pushed a commit: 0a2d33476f345a8f10eb47c9b32b03030953d355 for using the common protocol for all.